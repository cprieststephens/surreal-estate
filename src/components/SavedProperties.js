import React, { useState, useEffect } from "react";
import axios from "axios";
import SavedCard from "./SavedCard";
import Alert from "./Alert";
import "../styles/saved-properties.css";

const SavedProperties = () => {
  const [properties, setProperties] = useState([]);
  const [alert, setAlert] = useState({ message: "" });

  useEffect(() => {
    axios
      .get(`http://localhost:3000/api/v1/Favourite?populate=propertyListing`)
      .then(({ data }) => setProperties(data))
      .catch(() =>
        setAlert({
          message: "Server error. Please try again later.",
        })
      );
  }, [properties]);

  return (
    <div className="saved-properties">
      <div className="saved-properties__cards">
        {properties.map((property) => (
          <SavedCard
            key={property._id}
            {...property}
            setProperties={setProperties}
          />
        ))}
      </div>
      <div className="properties__alert">
        <Alert message={alert.message} />
      </div>
    </div>
  );
};

export default SavedProperties;
