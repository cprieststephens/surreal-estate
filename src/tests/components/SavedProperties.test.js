import React from "react";
import { render, waitFor } from "@testing-library/react";
import axios from "axios";
import SavedProperties from "../../components/SavedProperties";

describe("SavedProperties", () => {
  it("renders correctly", async () => {
    const mockData = {
      properties: [
        {
          id: "1111",
          propertyListing: {
            id: "2222",
            title: "stub-title",
            type: "stub-type",
            bedrooms: "stub-bedrooms",
            bathrooms: "stub-bathrooms",
            price: "stub-price",
            city: "stub-city",
            email: "stub-email",
          },
        },
      ],
    };
    jest.spyOn(axios, "get").mockResolvedValue(mockData);
    const { asFragment } = render(<SavedProperties />);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
