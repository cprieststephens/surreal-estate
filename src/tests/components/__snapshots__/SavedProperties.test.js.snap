// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedProperties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="saved-properties"
  >
    <div
      class="saved-properties__cards"
    />
    <div
      class="properties__alert"
    />
  </div>
</DocumentFragment>
`;
